### PR TITLE
test(core-http-utils): 100% coverage

### PR DESCRIPTION
**Pull request by jdar**
Unit tests of core-http-utils, per #1763

## What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactor
- [ ] Performance
- [x] Tests
- [ ] Build
- [ ] Documentation
- [ ] Code style update
- [ ] Continuous Integration
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Does this PR release a new version?

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch
- [x] All tests are passing
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by jdar** [comment]
Two notes: 

I could have created a mock-plugin to test the handling of plugins in create-server.test.ts
I could have mocked the @hapi server object

I looked around through the other unit tests, but didn't see that level of rigor, so I didn't.

I'm new to jest/.ts, so feedback appreciated.

File: __tests__/unit/core-http-utils/create-secure-server.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        createSecureServer(
+            { host: "0.0.0.0", port: 3000 },
+            theCallback,
+            { key: smallFilePath + ".key", cert: smallFilePath + ".pem", port: "4007" },
+            [require("hapi-rate-limit")],
+        );
+
+        // This tests functionality from createServer, which theoretically create duplicative error messages.
+        // The right way to resolve this, IMO, is to make a plugin which makes the server secure, rather than having createSecureServer()
+        theCallback();
```
Comment (jdar) [inline __tests__/unit/core-http-utils/create-secure-server.test.ts]: I'm not really sure what to test here. 

I could mock createServer. But as noted in the comment, seems more reasonable to refactor away the method createSecureServer, and just use a plugin.

Comment (faustbrian) [reply __tests__/unit/core-http-utils/create-secure-server.test.ts]: https://github.com/ArkEcosystem/core/pull/2730

**Comment by jdar** [comment]
@faustbrian 
The util-function tests now reflect your refactor of createSecureServer.

BUT: To get the bounty, I still need to work out how to best test existing and future plugins.
I tried to put together a unit test for plugins that mocks the server object, rather than the content-type plugin approach that uses the Hapi-Server and moreover re-testing the createServer method.

I'd appreciate your review of the general approach, before I continue down this path.
https://github.com/jdar/core/blob/04aec22b27a8ea5b8d9e3c7a02c80480fef06eb3/__tests__/unit/core-http-utils/plugins/hapi-ajv.test.ts#L41

It looks less clean, for sure, but on the other hand testing a validator is just going to be messier than testing the response of Content-Type arguments. And future plugins might interact with each other ... middleware-like?

Anyway. give it an eyeball, and let me know if you'd prefer I just try to use the createServer non-mocked approach you outlined in content-type.test.ts

**Comment by faustbrian** [comment]
My general stance on mocks is `Don't Mock What You Don't Own` so I try to avoid mocking any third-party dependencies, even mocking your own things I see as a code smell and mocking will be drastically reduced in Core 3.0.

With hapi that can be avoided very easily because you can spin up a server and inject requests instead of having to send an actual HTTP request with a client so I would always go with testing against a real instance of the third-party dependency instead of creating mocks which give you false confidence by making tests green.

https://hapijs.com/api#-await-serverinjectoptions

**Comment by faustbrian** [comment]
@jdar are you intending to complete this within the next days or what is the current status on this PR?

**Comment by jdar** [comment]
@faustbrian ... Will take name off ticket. Could not figure a robust mocking

**Pull request closed without merging**